lib/chezmoi/scripts.go

```diff
@@ -0,0 +1,93 @@
+package chezmoi
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+)
+
+var onceRegex = regexp.MustCompile(`(?m)^#\s*chezmoi:\s*once$`)
+
+// Script is a script supposed to run.
+type Script struct {
+	Name             string
+	SourcePath       string
+	once             bool
+	alreadyExecuted  bool
+	contents         []byte
+	contentsErr      error
+	evaluateContents func() ([]byte, error)
+}
+
+// Apply executes the script if necessary to reach target state.
+func (s *Script) Apply(destDir string, dryRun bool) error {
+	if s.once && s.alreadyExecuted {
+		return nil
+	}
+
+	s.Contents()
+	if s.contentsErr != nil {
+		return s.contentsErr
+	}
+
+	if err := s.parse(); err != nil {
+		return err
+	}
+	fmt.Printf("chezmoi: Running script %s\n", s.Name)
+	if dryRun {
+		println(string(s.contents), "\n")
```

Comment:
Please don't use the `println` function, it is a leftover from an old version of go. Use `fmt.Println` instead, and only print if running in verbose mode.